src/transformState.js

```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const deleteKey of action.keysToRemove) {
+          delete state[deleteKey];
+        }
+
+        break;
+
+      case 'clear':
+        for (const clear in state) {
+          delete state[clear];
+        }
+
+        break;
+
+      default:
+        break;
```

Comment:
Let's throw an error in case of `default` case

Reply:
Read more - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw